pkg/v14/kubeconfig/resource.go

```diff
@@ -18,9 +19,10 @@ const (
 // Config represents the configuration used to create a new kubeconfig resource.
 type Config struct {
 	// Dependencies.
-	CertSearcher certs.Interface
-	K8sClient    kubernetes.Interface
-	Logger       micrologger.Logger
+	CertSearcher  certs.Interface
+	K8sClient     kubernetes.Interface
+	Logger        micrologger.Logger
+	TransformFunc func(interface{}) (v1alpha1.ClusterGuestConfig, bool, error)
```

Comment:
Can we maybe come up with a better name? There is already some scheme for other places where we configure such key functions.

Reply:
Pawel once suggested `GetClusterConfigFunc` and I think it's better, WDYT?

Reply:
I just based it on the function that was already present in the code. It was `GetClusterConfig` IIRC. I think there are other places where we specify that in the codebase. I think it would make sense to make it consistent.